scalapb-runtime/shared/src/main/scala/scalapb/GeneratedMessageCompanion.scala

```diff
@@ -278,3 +278,10 @@ abstract class GeneratedFileObject {
   // Companion object of all top-level messages
   def messagesCompanions: Seq[GeneratedMessageCompanion[_]]
 }
+
+trait GeneratedSealedOneOf extends Any with Product with Serializable {
+  type MessageType = GeneratedMessage with Message[MessageType]
```

Comment:
Can we leave this abstract and leave this as an upperbound?

Reply:
That was my initial intent however I don't know how that could be done without introducing type parameters to the trait which then defeats the purpose of having the trait altogether because you would basically end up having to specify the type when matching against it. So instead of being able to simply write something like this:

`case value : scalapb.GeneratedSealedOneOf => value.asMessage`

One would have to write

```
case value : scalapb.GeneratedSealedOneOf[SomeMessageType1] => value.asMessage
case value : scalapb.GeneratedSealedOneOf[SomeMessageType2] => value.asMessage
case value : scalapb.GeneratedSealedOneOf[SomeMessageType3] => value.asMessage
case value : scalapb.GeneratedSealedOneOf[SomeMessageType4] => value.asMessage
case value : scalapb.GeneratedSealedOneOf[SomeMessageType5] => value.asMessage
...
```
I may be mistaken, but I'm unaware of a method to to add an upper bound to a type without introduces type parameters or generics.  Thoughts?

Reply:
Wouldn't `type MessageType <: GeneratedMessage with Message[MessageType]` work?

Reply:
I must've miss-typed something when I tried that and concluded it didn't work earlier. This should work. Thanks for the note. I'll have this updated tomorrow.

With that said, any chance we can get a new release soon with these changes included? I'd much rather rely on a full release vs a local build we reference in projects.

Thanks!